pkg/parser/grok/grok.go

```diff
@@ -0,0 +1,164 @@
+package grok
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/entry"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper"
+	"github.com/vjeantet/grok"
+)
+
+const operatorType = "grok_parser"
+
+func init() {
+	operator.Register(operatorType, func() operator.Builder { return NewConfig() })
+}
+
+// NewConfig creates a new grok parser config with default values
+func NewConfig() *Config {
+	return NewConfigWithID(operatorType)
+}
+
+// NewConfigWithID creates a new grok parser config with default values
+func NewConfigWithID(operatorID string) *Config {
+	return &Config{
+		ParserConfig: helper.NewParserConfig(operatorID, operatorType),
+	}
+}
+
+// Config is the configuration of a grok parser operator.
+type Config struct {
+	helper.ParserConfig `mapstructure:",squash"`
+
+	// grok pattern
+	Pattern string `mapstructure:"pattern"`
+
+	Cache struct {
+		Size uint16 `mapstructure:"size"`
+	} `mapstructure:"cache"`
+
+	// array of keys that will be present in the final values
+	Include []string `mapstructure:"include"`
+
+	// array of keys that will be removed from the final values
+	Exclude []string `mapstructure:"exclude"`
+}
+
+// Build will build a grok parser operator.
+func (c Config) Build(logger *zap.SugaredLogger) (operator.Operator, error) {
+	parserOperator, err := c.ParserConfig.Build(logger)
+	if err != nil {
+		return nil, err
+	}
+
+	if c.Pattern == "" {
+		return nil, fmt.Errorf("missing required field 'grok'")
```

Comment:
there is no grok field in the config?
```suggestion
		return nil, fmt.Errorf("missing required field 'pattern'")
```